pxf/pxf-json/src/test/java/org/apache/hawq/pxf/plugins/json/JsonExtensionTest.java

```diff
@@ -0,0 +1,288 @@
+	}
+
+	@Test(expected = IllegalStateException.class)
+	public void testMissingArrayJsonAttribute() throws Exception {
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		columnDefs.clear();
+
+		columnDefs.add(new Pair<String, DataType>("created_at", DataType.TEXT));
+		// User is not an array! An attempt to access it should throw an exception!
+		columnDefs.add(new Pair<String, DataType>("user[0]", DataType.TEXT));
+
+		List<String> output = new ArrayList<String>();
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-with-missing-text-attribtute.json"), output);
+	}
+
+	@Test
+	public void testMissingJsonAttribute() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		// Missing attributes are substituted by an empty field
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,,SpreadButter,tweetCongress,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-with-missing-text-attribtute.json"), output);
+	}
+
+	@Test
+	public void testMalformedJsonObject() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add(",,,,,,"); // Expected: malformed json records are transformed into empty rows
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-broken.json"), output);
+	}
+
+	@Test
+	public void testSmallTweets() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547123646465,text2,patronusdeadly,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+		output.add("Fri Jun 07 22:45:03 +0000 2013,343136551322136576,text4,SevenStonesBuoy,,-6.1,50.103");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-small.json"), output);
+	}
+
+	@Test
+	public void testTweetsWithNull() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,,text2,patronusdeadly,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/null-tweets.json"), output);
+	}
+
+	@Test
+	public void testSmallTweetsWithDelete() throws Exception {
```

Comment:
Shows that the simple (e.g. LineRecordReader) mode can handle missing or null json attributes. Those are converted into empty fields.